### PR TITLE
Finalize build, versatile config, better properties, more tests

### PR DESCRIPTION
**Pull request by golyshev**
- add client.finalize_build method (with test)
- refactor all properties on config to be more versatile (+ setters and tests)
- make sure that client can access config object passed in the constructor

@fotinakis

### CODE REVIEW

File: percy/config.py [hunk 1]
```diff
@@ -6,18 +6,36 @@ class AuthError(Exception):
 
 
 class Config(object):
+    
+    def __init__(self, api_url='https://percy.io/api/v1', default_widths=[], access_token=None):
```
Comment (fotinakis) [inline percy/config.py]: Style: avoid setting mutable objects as default arguments, by doing `default_widths=[]` the list is scoped outside of the method and can be accidentally mutated. Ie:

```python
>>> def foo(bar=[]):
...   bar.append(1)
...   print bar
... 
>>> foo()
[1]
>>> foo()
[1, 1]
>>> foo()
[1, 1, 1]
>>> foo()
[1, 1, 1, 1]
>>> foo()
[1, 1, 1, 1, 1]
>>> 
```

So just do `=None` instead and handle the default inside the method.

File: percy/config.py [hunk 2]
```diff
@@ -6,18 +6,36 @@ class AuthError(Exception):
...
-            raise AuthError('You must set PERCY_TOKEN to authenticate.')
-        return access_token
+        if not self._access_token:
+            raise AuthError('You must set PERCY_TOKEN to authenticate.')        
+        return self._access_token
+
+    @access_token.setter
+    def access_token(self, value):
+        self._access_token = value
+        if not self._access_token:
```
Comment (fotinakis) [inline percy/config.py]: Drop this check since we have one in the getter.

File: percy/resource.py [hunk 3]
```diff
@@ -9,9 +9,11 @@ def __init__(self, resource_url, is_root=False, **kwargs):
         self.resource_url = resource_url
         if 'sha' in kwargs and 'content' in kwargs or not ('sha' in kwargs or 'content' in kwargs):
             raise ValueError('Exactly one of sha or content is required.')
+        # Encode the content if it exists
+        self.content = kwargs['content'].encode('utf-8') if kwargs.get('content') else None
```
Comment (fotinakis) [inline percy/resource.py]: Let's re-evaluate this encoding handling later—we need to be able to upload binary resources like images as well.

File: tests/test_client.py [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
 import percy
+from percy import config
 
 
 class TestPercyClient(unittest.TestCase):
 
     def setUp(self):
         os.environ['PERCY_TOKEN'] = 'abcd1234'
-        self.percy_client = percy.Client()
+        self.percy_client = percy.Client(config=config.Config(default_widths=(1024, 768)))
```
Comment (fotinakis) [inline tests/test_client.py]: Nitpick: change `(1024, 768)` to `(1280, 375)` for consistency. They are actually just width breakpoints, but using 1024 and 768 makes it looks like a screen size.

**Comment by fotinakis** [comment]
Looks great @golyshev, just a couple of comments above.

File: percy/resource.py (hunk 3 above)
Comment (golyshev) [inline percy/resource.py]: Yeah, this problem manifests only in python 3 (plus there are some other issues with JSON encoder) - see travis

**Pull request merged**